src/test/java/spoon/test/variable/VariableTest.java

```diff
@@ -37,6 +44,64 @@
 
 public class VariableTest {
 
+    @Test
+    public void test() {
+        // contract: one can use setType without having a very hard generics type checking error
+
+        // to get this done, we use the refactoring below
+        CtClass c = Launcher.parseClass("class C{ void f() {Object i=null; }}");
+        List<CtLocalVariable> localVariables = c.getElements(new TypeFilter<>(CtLocalVariable.class));
+        System.out.println(localVariables);
+
+        // before: "CtLocalVariable<?> lv" could not typecheck, with a hard generics error
+        // now it works; we can use both "CtLocalVariable<?> lv" and "CtLocalVariable lv"
+        // it's source level backward compatible for Spoon itself
+        CtLocalVariable<?> lv = localVariables.get(0);
+        lv.setType(c.getFactory().Type().createReference(Object.class));
+
+        CtLocalVariable lv2 = localVariables.get(0);
+        lv2.setType(c.getFactory().Type().createReference(Object.class));
+
+    }
+
+    @Test
+    public void refactor() {
+        // rectoring all setType(CtTypeParameter<T>) -> setType(CtTypeParameter)
```

Comment:
I am not sure if I am a huge fan of a test overwriting *the very source code I am currently editing*. That sounds a bit scary.

What exactly is the purpose of this test? Should it enforce that nobody re-introduces this pattern? In that case, I would probably make this a model test and just *verify*, without printing anything. We have a few of such tests checking various properties of spoon. The test could then be something along these lines:
```java
@ModelTest("src/main/java")
void ensureNoWildcardUsedForSetType...(Model model) {
  model.processWith(new Abstract....() {
    if (isToBeProcessed()) { assertFalse("message"); }
  })
}
```

If we want to automatically refactor it continuously (similar to the CloneBuilder CtGeneration tests), we might want to think about using a similar pattern.

If this is a one-off, I am not sure the test is worth including in master. Maybe it can live in this PR as documentation instead?

Reply:
Fully agree, removed @Test